client/api_client.js

```diff
@@ -10,9 +10,26 @@ var restClient = require('request-promise'),
  * @constructor
  */
 function ApiClient(urlBase, serviceId, apiKeyId) {
-  this.serviceId = serviceId;
-  this.apiKeyId = apiKeyId;
-  this.urlBase = urlBase;
+  if (arguments.length === 1 && !arguments[0].startsWith('http')) {
+    this.urlBase = 'https://api.notifications.service.gov.uk';
+    serviceId = arguments[0];
+  } else {
+    this.urlBase = urlBase;
+  }
+  if (arguments.length === 2 && !arguments[0].startsWith('http')) {
```

Comment:
i don't like this super flexible permissive API - it makes this function very awkward and leaves a lot of ambiguity about what each parameter actually means.

I'd prefer to have these strict. in an ideal world i'd say make them part of an options dict you pass in since that's easy to reason about what they intended to supply, but we're here now and if we dont want to break backwards compatibility we're stuck with the existing params layout. i'm fine enforcing that they pass in a parameter.

could always allow users to pass in ApiClient(null, my_api_key) and we check for null to put in the default url. idk.

this code is hella gnarly tho